crates/fuel-streams/src/stream/stream_impl.rs

```diff
@@ -185,15 +228,64 @@ impl<S: Streamable> Stream<S> {
     /// # Ok(())
     /// # }
     /// ```
-    pub async fn subscribe_with_config(
-        &self,
+    pub async fn subscribe_with_config<'a>(
+        &'a self,
         opts: StreamConfig,
-    ) -> Result<PullConsumerStream, StreamError> {
+    ) -> Result<Pin<Box<dyn futures::Stream<Item = S> + Send + 'a>>, StreamError>
+    {
         self.stream
-            .subscribe_consumer(SubscribeConsumerConfig {
+        // TODO: Improve DX by ensuring the stream returns the streamable entity directly
+            .subscribe(Some(SubscriptionConfig {
+                deliver_policy: opts.deliver_policy,
+                filter_subjects: self.filter_subjects.to_owned(),
+            }))
+            .await
+            .map_err(|source| StreamError::SubscribeWithOpts { source })
+    }
+
+    /// Subscribes to the stream bytes with custom configuration options.
+    ///
+    /// # Parameters
+    ///
+    /// * `opts`: A `StreamConfig` instance containing custom configuration options.
+    ///
+    /// # Returns
+    ///
+    /// Returns a `Result` containing a `PullConsumerStream` on success,
+    /// or a `StreamError` on failure.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use fuel_streams::types::FuelNetwork;
+    /// use fuel_streams::client::Client;
+    /// use fuel_streams::stream::{Stream, StreamConfig};
+    /// use fuel_streams::blocks::Block;
+    /// use fuel_streams::types::DeliverPolicy;
+    ///
+    /// # async fn example() -> Result<(), fuel_streams::Error> {
+    /// # let client = Client::connect(FuelNetwork::Local).await?;
+    /// # let stream = Stream::<Block>::new(&client).await;
+    /// let config = StreamConfig {
+    ///     deliver_policy: DeliverPolicy::All,
+    /// };
+    /// let subscription = stream.subscribe_with_config(config).await?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub async fn subscribe_raw_with_config<'a>(
```

Comment:
We should consider whether external users really need to subscribe for raw bytes. For now, it might be better to keep this method internal to avoid bloating the SDK. It could be better to decode internally in the SDK and deliver the final typed data instead.